src/main/scala/is/hail/expr/AST.scala

```diff
@@ -583,17 +579,33 @@ case class ApplyMethodAST(posn: Position, lhs: AST, method: String, args: Array[
             AggOp.fromString.lift(method))
           bodyx <- body.toIR()
           aggSig = AggSignature(op,
-            bodyx.typ,
             FastIndexedSeq(),
             None,
-            FastIndexedSeq())
+            FastIndexedSeq(bodyx.typ))
           ca <- fromOption(
             this,
             "no CodeAggregator",
             AggOp.getOption(aggSig))
           rx <- lhs.toAggIR(agg.get, x => ir.SeqOp(ir.I32(0), FastIndexedSeq(ir.Let(name, x, bodyx)), aggSig))
         } yield
           ir.ApplyAggOp(rx, FastIndexedSeq(), None, aggSig): IR
+      case (t: TAggregable, "count", IndexedSeq()) =>
+        for {
+          op <- fromOption(
+            this,
+            s"no AggOp for method $method",
+            AggOp.fromString.lift(method))
+          aggSig = AggSignature(op,
+            FastIndexedSeq(),
+            None,
+            FastIndexedSeq())
+          ca <- fromOption(
+            this,
+            "no CodeAggregator",
+            AggOp.getOption(aggSig))
+          rx <- lhs.toAggIR(agg.get, x => ir.SeqOp(ir.I32(0), FastIndexedSeq(), aggSig))
```

Comment:
Tests haven't run yet, but is this going to work?  SeqOp still contains the dummy Int32, but elsewhere you changed things to (correctly) have no seqOp args for collect.  I would also expect you need to change the Python agg.count.

Reply:
Oops, misread, that's the aggregator number.  We can change the Python, but this is fine for now.

Reply:
It doesn't have the dummy i32. The first argument is the aggregator index. I'm still thinking about the Python agg.count

Reply:
It looks like the Python implementation doesn't matter for this.

Reply:
The count in the function registry takes a Int64 (which Python generates) but it's discarded by toIR.  It can get ripped out before toIR, but that doesn't seem urgent.

Reply:
Ah, I see, it isn't easy to rip out the argument before toIR because the old code requires one (the parameter of the TAggregable).  All the more reason to rip out the AST!